packages/astro/src/core/render-context.ts

```diff
@@ -89,34 +89,38 @@ export class RenderContext {
 		const apiContext = this.createAPIContext(props);
 		const { type } = routeData;
 
-		const lastNext =
-			type === 'endpoint'
-				? () => renderEndpoint(componentInstance as any, apiContext, serverLike, logger)
-				: type === 'redirect'
-					? () => renderRedirect(this)
-					: type === 'page'
-						? async () => {
+		const lastNext = async () => {
+			if (routeData.type === 'endpoint') return renderEndpoint(componentInstance as any, apiContext, serverLike, logger);
```

Comment:
We should use a `switch (routeData.type)` here. If we add a new variant to `type`, TS and ESlint will be both errors and prevent the build.